Kitodo/src/main/java/org/kitodo/production/model/Subfolder.java

```diff
@@ -347,7 +347,7 @@ private Map<String, URI> listDirectory(Pair<URI, Pattern> query, boolean absolut
         FilenameFilter filter = (dir, name) -> query.getRight().matcher(name).matches();
         try (Stream<URI> relativeURIs = fileService.getSubUris(filter, query.getLeft()).parallelStream()) {
             Stream<URI> resultURIs = absolute ? relativeURIs.map(
-                uri -> new File(FilenameUtils.concat(ConfigCore.getKitodoDataDirectory(), uri.getPath())).toURI())
+                uri -> new File(ConfigCore.getKitodoDataDirectory().concat(uri.getPath())).toURI())
```

Comment:
Question: Why did you change this?

Reply:
This was a tricky one. `FilenameUtils.concat` returns `null` for relative paths. And since the Kitodo data directory path is relative in the Selenium test environment, this led to `NullPointerException`s and made it impossible to retrieve the actual metadata directory of a specific process during Selenium tests. The `File.concat` function on the other hand can handle relative paths correctly, so I decided to use it here.